src/main/java/space/obminyashka/items_exchange/dao/AdvertisementRepository.java

```diff
@@ -30,6 +30,9 @@ public interface AdvertisementRepository extends JpaRepository<Advertisement, UU
     @Query("SELECT a FROM Advertisement a WHERE LOWER(a.topic) IN :topics")
     Page<Advertisement> search(@Param("topics") Set<String> topics, Pageable pageable);
 
+    @Query("SELECT a FROM Advertisement a WHERE :categoryId is null or a.subcategory.category.id = :categoryId")
```

Comment:
Using subcategory I extracted the category id

Reply:
Didn't catch the case why should we extract the result for the null category?

Reply:
refacted